--- libqalculate/util.cc
+++ libqalculate/util.cc
@@ -1164,8 +1164,8 @@
 }
 
 void Thread::enableAsynchronousCancel() {
-	pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
-	pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
+	//pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
+	//pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
 }
 
 void *Thread::doRun(void *data) {
@@ -1184,7 +1184,8 @@
 }
 
 bool Thread::cancel() {
 	if(!running) return true;
-	running = pthread_cancel(m_thread) != 0;
+	//running = pthread_cancel(m_thread) != 0;
+	running = pthread_kill(m_thread, 0) != 0;
 	return !running;
 }
 
